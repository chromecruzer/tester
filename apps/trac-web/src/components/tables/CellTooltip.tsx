import React, {ReactElement} from "react";
import {Tooltip} from "react-tooltip";
import 'react-tooltip/dist/react-tooltip.css'
export const CellTooltip = ({message, children, cellProps, columnId, rowId}) : ReactElement => {
  // console.log('tooltip cell props', cellProps);
  // console.log('tooltip children', children);
  if(message) {
    const ttId = `${rowId}_${columnId}_tt`;
    cellProps.className += ' bg-bl-consumer-off-white text-bl-text-dark';
    return <td
      data-tip={true} data-for={ttId}
      {... cellProps}
    >
      {children}
      <Tooltip anchorId={ttId}>
        {message}
      </Tooltip>
    </td>
  }
  return <td {...cellProps}>{children}</td>;
};
