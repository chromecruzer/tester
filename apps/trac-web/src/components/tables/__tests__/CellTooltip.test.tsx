import {CellTooltip} from "../CellTooltip";
import {render, screen, within} from "@testing-library/react";

describe('CellTooltip', () => {
  let tdChild;
  beforeEach(() => {
    tdChild = 'I am a TD child';
  });
  const renderCellTooltip = (expectedMessage) => {
    const cellProps = {className: ''};
    return render(
      <table><tbody><tr>
      <CellTooltip
      message={expectedMessage}
      cellProps={cellProps}
      columnId='columnId'
      rowId='rowId'
  >
      {tdChild}
    </CellTooltip></tr></tbody></table>)
  }
  it('should create a tooltip with the td cell', () => {
    renderCellTooltip('Expected Tooltip');
    // screen.debug();
    const td = screen.queryByText(tdChild);
    // screen.debug(td);
    if(td === null) {
      throw new Error('Cannot find table cell');
    }
    expect(td.getAttribute('class')).toMatchSnapshot();
    expect(td.getAttribute('data-for')).toMatchSnapshot();
    expect(td.getAttribute('data-tip')).toBeTruthy();
    const tooltip = screen.queryByText('Expected Tooltip');
    const id = td.getAttribute('data-for');
    // screen.debug(tooltip);
    if(tooltip === null) {
      throw new Error('Cannot find tooltip');
    }
    expect(tooltip.getAttribute('id')).toEqual(id);
  });
  it('should create a normal td cell', () => {
    renderCellTooltip(null);
    // screen.debug();
    const td = screen.queryByText(tdChild);
    expect(td).not.toBeNull();
    const tooltip = screen.queryByText('Expected Tooltip');
    expect(tooltip).toBeNull();
  });
});
