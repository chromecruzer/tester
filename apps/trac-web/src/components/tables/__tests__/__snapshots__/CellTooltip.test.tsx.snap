// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellTooltip should create a tooltip with the td cell 1`] = `" bg-bl-consumer-off-white text-bl-text-dark"`;

exports[`CellTooltip should create a tooltip with the td cell 2`] = `"rowId_columnId_tt"`;
